linera-views/src/graphql.rs

```diff
@@ -10,6 +10,31 @@ use crate::{
 
 // TODO(#1326): fix the proliferation of constraints from this module
 
+// TODO(#1858): come up with a better name-mangling scheme
+/// Mangle a GraphQL type into something that can be interpolated into a GraphQL type name
+fn mangle(type_name: impl AsRef<str>) -> String {
+    let mut mangled = String::new();
+
+    for c in type_name.as_ref().chars() {
+        match c {
+            '!' | '[' => (),
```

Comment:
I was thinking we would to map `!` to `_NonNull`.

Reply:
^^ @Twey

Reply:
Oh sorry I missed this one!
I don't think it's important to distinguish nullable/non-nullable in the type name.  These names are only for human-readability (now) and I don't think it's likely that a user will confuse a nullable type for another type that is the same modulo nullability.

Reply:
Thanks. Sounds good!